shared/js/content-scripts/block-cookie.js

```diff
@@ -28,72 +28,109 @@
         document.__defineGetter__('cookie', () => '')
     }
 
-    function enforceCookieExpiry () {
-        var cookieSetter = document.__lookupSetter__('cookie')
-        // Enforce 7 day max expiry
-        document.__defineSetter__('cookie', function (value) {
-            const nextWeek = 604800
-            var comps = value.split(';')
-            var expiryFound = false
-            for (var i = 0; i < comps.length; i++) {
-                if (comps[i].toLowerCase().contains('session')) {
-                    expiryFound = true
-                    break
-                }
-
-                if (comps[i].trim().startsWith('expires')) {
-                    expiryFound = true
-                    const dateStr = comps[i].trim().replace('expires=', '').trim()
-                    const setDate = new Date(dateStr)
-                    const nextWeekDate = new Date()
-                    nextWeekDate.setTime(nextWeekDate.getTime() + nextWeek)
-                    if (setDate > nextWeekDate) {
-                        comps[i] = 'max-age=' + nextWeek
-                    }
+    /**
+     * Apply an expiry policy to cookies set via document.cookie.
+     * @param {string} secret Used to detect messages sent from the extension content-script
+     */
+    function applyCookieExpiryPolicy (secret) {
+        const cookieSetter = document.__lookupSetter__('cookie')
+        const cookieGetter = document.__lookupGetter__('cookie')
+        const lineTest = /(\()?(http[^)]+):[0-9]+:[0-9]+(\))?/
```

Comment:
Can you please check with @jonathanKingston how he is parsing the stack trace? I know he spent quite some time testing different variants in chrome and firefox (https://privacy-test-pages.glitch.me/features/stack-tracing/).
We should probably consolidate both methods at some point.

Reply:
This is the regex copied from his PR.

I'm not sure the best way to share code for this, given this is in a function injected into the page context via `Function.toString()`, so needs to be self contained.

Reply:
Sorry missed this, we could generate a file into the `content-scope` directory from a utils directory perhaps?